pkg/distributors/contracts/MultiDistributor.sol

```diff
@@ -494,25 +500,25 @@ contract MultiDistributor is IMultiDistributor, ReentrancyGuard, MultiDistributo
         for (uint256 i; i < stakingTokens.length; i++) {
             IERC20 stakingToken = stakingTokens[i];
             UserStaking storage userStaking = _userStakings[stakingToken][msg.sender];
-            unstake(stakingToken, userStaking.balance, msg.sender);
+            unstake(stakingToken, userStaking.balance, msg.sender, msg.sender);
         }
 
-        _claim(distributionIds, address(callbackContract), IVault.UserBalanceOpKind.TRANSFER_INTERNAL);
+        _claim(distributionIds, IVault.UserBalanceOpKind.TRANSFER_INTERNAL, msg.sender, address(callbackContract));
         callbackContract.distributorCallback(callbackData);
     }
 
-    function _stakeFor(
+    function _stake(
         IERC20 stakingToken,
         uint256 amount,
-        address user,
-        address from
+        address sender,
+        address recipient
     ) internal {
         require(amount > 0, "STAKE_AMOUNT_ZERO");
 
         // Before we increase the user's staked balance we need to update all of their subscriptions
```

Comment:
Fixed. I've left the final reference to "user" as it's speaking more generally rather than specific to the users involved in the current staking action.